src/App.js

```diff
@@ -1,16 +1,72 @@
-import React from 'react';
+import React, { Component } from 'react'
+import './components/TodoComponents/Todo.css'
+import TodoList from './components/TodoComponents/TodoList'
+import TodoForm from './components/TodoComponents/TodoForm'
 
-class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+class App extends Component {
+  inputElement = React.createRef()
+  constructor() {
+    super()
+    this.state = {
+      items: [],
+      currentItem: {
+        text: '',
+        key: '',
+        completed:false
+      },
+    }
+  }
+  deleteItem = key => {
+    var filteredItems = this.state.items.filter(item => {
+      return item.key === key
+
+    })
+    filteredItems [0].completed = true
+    this.setState({
+      items: this.state.items,
+    })
+  }
+  deleteAll = e => {
+    var filteredItems = this.state.items.filter(item => {
+      return !item.completed
```

Comment:
This is great! Well done using the `.filter()` method. That is the best way to delete multiple items out of state (assuming their is something common among them that you can filter by).